### PR TITLE
Java SDK section in Docusaurus

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
###  Description

- Buildout Java SDK section

### Staging

- [Entire Java SDK section](http://realm-io-staging.s3-website-us-east-1.amazonaws.com/nathan-contino-mongo/docs-realm/javaSDK/sdk/java/intro/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: docusaurus/docs/sdk/java/quick-start.mdx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+:::
+
+## Define Your Object Model
+
+Your application's **data model** defines the structure of data
+stored within Realm Database.
+You can define your application's data model via Kotlin or
+Java classes in your application code with
+:ref:`Realm Object Models <create-schema-from-rom>`.
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/quick-start.mdx:93]: this an oversight or waiting for that page to be made?

File: docusaurus/docs/sdk/java/quick-start.mdx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    <TabItem value="java" label="Java">
+         <CodeBlock language="java">{OpenARealmJava}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+## Create, Read, Update, and Delete Objects
+
+Once you have opened a realm, you can modify the
+objects within that realm in a write transaction.
+
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/quick-start.mdx:128]:
```suggestion

### Create an Object

```

Comment (mongodben) [reply docusaurus/docs/sdk/java/quick-start.mdx:128]: add other H3s as relevant for other examples here

File: docusaurus/docs/sdk/java/quick-start.mdx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    </TabItem>
+    <TabItem value="java" label="Java">
+         <CodeBlock language="java">{DeleteJava}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+## Watch for Changes
+
+You can :ref:`watch a realm, collection, or object for changes
+<java-client-notifications>` by attaching a custom
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/quick-start.mdx:189]: ref should be link

File: docusaurus/docs/sdk/java/quick-start.mdx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    </TabItem>
+    <TabItem value="java" label="Java">
+         <CodeBlock language="java">{WatchJava}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+## Complete Example
+
+If you're running this project in a fresh Android Studio project, you can
+copy and paste this file into your application's `MainActivity` -- just
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/quick-start.mdx:205]:
```suggestion
copy and paste this file into your application's `MainActivity`. Just
```

File: docusaurus/docs/sdk/java/android-livedata.mdx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+## Prerequisites
+
+This quick start guide uses Sync to synchronize data
+changes between clients. Before you begin, ensure you have:
+
+- [Installed the Java SDK](/sdk/java/install).
+- [Created a Realm App](https://docs.mongodb.com/realm/manage-apps/create/create-with-realm-ui/#std-label-create-a-realm-app).
+- [Enabled anonymous authentication](https://docs.mongodb.com/realm/authentication/anonymous/#std-label-anonymous-authentication-configuration).
+- [Enabled Sync](https://docs.mongodb.com/realm/sync/configure/enable-sync/#std-label-enable-sync).
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/android-livedata.mdx:30]: remove periods at end of bullet points

File: docusaurus/docs/sdk/java/realm-database/query-language.mdx [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+`Latin Extended B (UTF-8 range 0–591)` character sets. Setting
+the case insensitive flag in queries when using `equalTo`,
+`notEqualTo`, `contains`, `endsWith`, `beginsWith`, or
+`like` only works on English locale characters.
+
+## Aggregate Operators
+
+You can apply an aggregate operator to a collection property
+of a Realm object. Aggregate operators traverse a
+:ref:`collection <java-client-collections>` and reduce it
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/realm-database/query-language.mdx:386]: replace w link

File: docusaurus/docs/sdk/java/realm-database/filter-data.mdx [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+The `where()` method returns a `RealmQuery` that you can resolve into
+a `RealmResults` using a `find` method. Filtered results can only
+return objects of the same type as the original results set.
+
+## Query with Realm Query Language
+
+Use [`realmQuery.rawPredicate()`](https://docs.mongodb.com/realm-sdks/java/latest/io/realm/RealmQuery.html#rawPredicate-java.lang.String-) to query
+realms using [Realm Query Language](/sdk/java/realm-database/query-language),
+a language inspired by
+`NSPredicate <https://academy.realm.io/posts/nspredicate-cheatsheet/>`.
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/realm-database/filter-data.mdx:228]: fix link

File: docusaurus/docs/sdk/java/realm-database/realms/open-and-close-a-realm.mdx [hunk 8]
```diff
@@ -0,0 +1,213 @@
+---
```
Comment (mongodben) [inline docusaurus/docs/sdk/java/realm-database/realms/open-and-close-a-realm.mdx:1]: there's a lot of stuff on this page that needs md-ification

File: docusaurus/docs/sdk/java/quick-start.mdx (hunk 2 above)
Comment (nathan-contino-mongo) [reply docusaurus/docs/sdk/java/quick-start.mdx:127]: Not sure I understand -- this section has each of those examples in it?

Comment (nathan-contino-mongo) [reply docusaurus/docs/sdk/java/quick-start.mdx:127]: Added

**Review by mongodben — APPROVED** [review]
approving this PR. wasn't able to review every single thing before have to leave for day (made it til [this page](http://realm-io-staging.s3-website-us-east-1.amazonaws.com/nathan-contino-mongo/docs-realm/javaSDK/sdk/java/realm-database/read/iteration/)). approving b/c this page isn't going to the main site and we need to build on top of the merged PR tomorrow. 

the whole android section should be reviewed before we go live. i can continue on this later.

**Pull request merged**